src/chainstate/coordinator/mod.rs

```diff
@@ -683,6 +996,42 @@ impl<
         )
     }
 
+    // If it is time to process the exit-at-rc contract for a specific reward cycle, this function
+    // returns that reward cycle. Otherwise, the function returns None.
+    fn process_exit_reward_cycle(
+        &mut self,
+        stacks_block_height_in_cycle: u64,
+        parent_exit_info: &BlockExitRewardCycleInfo,
+    ) -> Result<Option<BlockExitRewardCycleInfo>, Error> {
+        info!(
+            "RCPR: curr height: {}, parent height: {}",
+            stacks_block_height_in_cycle, parent_exit_info.stacks_block_height_in_cycle
+        );
+        if stacks_block_height_in_cycle == 2 {
+            // We want to process the exit-at-rc contract reward cycle for the reward cycle the
+            // current block's parent's parent belongs to since the height (in number of Stacks
+            // blocks) of the current block in its reward cycle is 2.
+            let exit_info_opt = SortitionDB::get_exit_at_reward_cycle_info(
+                self.sortition_db.conn(),
+                &parent_exit_info.parent_block_id,
+            )?;
+            Ok(exit_info_opt)
+        } else if stacks_block_height_in_cycle == 1
+            && parent_exit_info.stacks_block_height_in_cycle == 1
+        {
+            // This is an edge case in which the previous reward cycle contains only a single Stacks block.
+            // It is time to process the exit-at-rc contract for the reward cycle the current block's
+            // parent's parent belongs to.
```

Comment:
If there's a reward cycle that contains only a single Stacks block, it might simplify things to simply declare that a vote tally will not occur.  In such conditions, the network is already well and truly dead anyway.

Reply:
The processing occurs for the parent cycle (which could be any length) so not sure if we should introduce that rule.